### ISSUE TITLE
run-time panic when declaring non-sensical layout constraint

### ISSUE THREAD

**Issue by tronical** [slint-ui/slint#257]
The following test case is the result of reducing a larger (valid) design where a `width` was applied to a vertical layout, which in turn is in a horizontal layout, nested in more layouts.

```
export TestCase := Rectangle {
    VerticalLayout {
        HorizontalLayout {
            VerticalLayout {
                width: parent.width;
                Rectangle {}
            }
        }
    }
}
```

This produces the `Recursion detected` panic in the property system.

Ideally the compiler should detect this circular dependency.

**Comment by ogoffart** [comment]
This no longer call a runtime panic.

But if we preview that, there is a loop that makes the window grow endlessly.
That's because the compile-time loop detection do not see the fact that the Window's size depends on the Windows's minimum-size and it maximum-size, indirectly through the window manager

**Comment by ogoffart** [comment]
Actually, this has recently been reported as https://github.com/slint-ui/slint/issues/2902
it has a better description, so i'll close this one as duplicated

**Comment by ogoffart** [comment]
closing as duplicate of https://github.com/slint-ui/slint/issues/2902
